aws/config.go

```diff
@@ -86,3 +86,19 @@ func (c Config) Copy() Config {
 	cp := c
 	return cp
 }
+
+// EndpointDiscoveryEnableState indicates if endpoint discovery is
+// enabled, disabled, or in default (auto) behavior state.
+//
+// Default behavior (AUTO) indicates operations that require endpoint
+// discovery will use Endpoint Discovery by default. Operations that
+// optionally use Endpoint Discovery will not use Endpoint Discovery
+// unless EndpointDiscovery is explicitly enabled.
+type EndpointDiscoveryEnableState string
+
+// Enumeration values for EndpointDiscoveryEnableState
+const (
+	EndpointDiscoveryAuto     = EndpointDiscoveryEnableState("auto")  // default state
+	EndpointDiscoveryDisabled = EndpointDiscoveryEnableState("false") // endpoint discovery disabled
+	EndpointDiscoveryEnabled  = EndpointDiscoveryEnableState("true")  // endpoint discovery enabled
+)
```

Comment:
I am fine with using ``string`` or ``uint``. I followed [feedback as per comment](https://github.com/aws/aws-sdk-go-v2/pull/1263#discussion_r634747199) which was suggested to avoid pointer value representation. 

With ``uint`` we would need the extra Unset enum, and have a minor optimization of using less bytes for ``uint`` vs ``string``.

Reply:
Thinking again,  using ``uint`` would need us to export both ``Unset`` and ``Auto`` as configuration values. However both have the same behavior. 

Keeping this string but open to debate/thoughts.